src/web/src/components/SearchBar.tsx

```diff
@@ -33,107 +35,114 @@ const useStyles = makeStyles((theme: Theme) =>
+        backgroundColor: '#EEEEEE',
       },
       '& * > .MuiSvgIcon-root': {
         fontSize: '2rem',
-        color: theme.palette.primary.contrastText,
       },
       margin: 0,
       position: 'absolute',
-      right: '10px',
+      left: '10px',
       top: '6px',
     },
-    selectControl: {
-      '& > *': {
-        width: 'auto',
-        transform: 'translateY(2px)',
-        fontSize: '1.5rem',
-        textTransform: 'capitalize',
-        color: theme.palette.primary.main,
-        paddingLeft: '2rem',
-        [theme.breakpoints.down('xs')]: {
-          paddingLeft: '1rem',
-          width: '10rem',
-          transform: 'translateY(15px)',
-        },
+    clearIcon: {
+      color: '#999999',
+      '&:hover': {
+        backgroundColor: '#EEEEEE',
       },
-    },
-    selectItem: {
-      fontSize: '1.4rem',
-      textTransform: 'capitalize',
-      color: theme.palette.primary.main,
+      '& * > .MuiSvgIcon-root': {
+        fontSize: '3rem',
+      },
+      margin: 0,
+      position: 'absolute',
+      right: '5px',
+      top: '1px',
     },
   })
 );
 
 const SearchBar = () => {
   const classes = useStyles();
+  const [keyword, setKeyword] = useState('');
+
+  const [advancedSearchButtonVisible, setAdvancedSearchButtonVisible] = useState(false);
 
   const { filter, onFilterChange, onSubmitHandler } = useSearchValue();
+  const router = useRouter();
 
-  const searchOptions = ['post', 'author'];
+  const handleSearch = () => {
+    if (filter) {
+      if (filter.toUpperCase() === 'AUTHOR') {
+        router.push(`/search?text=${keyword}&filter=author`);
+      } else {
+        router.push(`/search?text=${keyword}&filter=post`);
+      }
+    } else {
+      router.push(`/search?text=${keyword}&filter=post`);
+    }
+  };
 
   return (
     <Box className={classes.root}>
-      <Paper component="form" className={classes.card} elevation={0}>
-        <Grid container direction="row" spacing={2} alignItems="center" justify="flex-start">
-          <Grid item xs={12} sm={2} lg={2}>
-            <FormControl fullWidth>
-              <TextField
-                id="standard-select-search-type"
-                select
-                value={filter}
-                InputProps={{ disableUnderline: true }}
-                className={classes.selectControl}
-                onChange={(event) => onFilterChange(event.target.value)}
-              >
-                {searchOptions.map((option) => (
-                  <MenuItem key={option} value={option} className={classes.selectItem}>
-                    {option}
-                  </MenuItem>
-                ))}
-              </TextField>
-            </FormControl>
-          </Grid>
-          <Grid item xs={12} sm={10} lg={10}>
-            <FormControl fullWidth>
-              <SearchInput />
+      <FormControl fullWidth>
+        <Grid item xs={12} sm={10} lg={10}>
+          <form
+            onSubmit={(e) => {
+              e.preventDefault();
```

Comment:
replace with onSubmitHandler(e)

The function already does the `e.preventDefault()` so you don't need that there